src/project-manager.ts

```diff
@@ -446,16 +475,57 @@ export class ProjectConfiguration {
     private fs: InMemoryFileSystem;
     private configFileName: string;
     private configContent: any;
+    private packageJsonFileName: string;
+
+    dtsNames: any;
+
 
     /**
      * @param fs file system to use
      * @param configFileName configuration file name (relative to workspace root)
      * @param configContent optional configuration content to use instead of reading configuration file)
+     * @param packageJsonFileName optional package.json file (relative to workspace root)
      */
-    constructor(fs: InMemoryFileSystem, configFileName: string, configContent?: any) {
+    constructor(fs: InMemoryFileSystem, configFileName: string, configContent?: any, packageJsonFileName?: string) {
         this.fs = fs;
         this.configFileName = configFileName;
         this.configContent = configContent;
+        this.packageJsonFileName = packageJsonFileName;
+        this.dtsNames = {};
+    }
+
+    /**
+     * Setter for package.json file for project configuration
+     * @param packageJsonFileName package.json file (relative to workspace root)
+     */
+    public setPackageJsonFile(packageJsonFileName: string) {
+        this.packageJsonFileName = packageJsonFileName;
+    }
+
+    /**
+     * Finds d.ts files located in /tmp/DefinitelyTyped folder which are related to dependencies from package.json
+     * @param deps list of dependecies from package.json file
+     */
+    public processDeps(deps) {
+        if (deps) {
+            for (let dep in deps) {
+                let findRes = findfiles(`/tmp/DefinitelyTyped/${dep}`, {
```

Comment:
What type is `deps`? Is it an array or object? Can you add a TypeScript type? Also, `for (let x in y) { ... }` does not have consistent ordering for arrays, so it is not recommended to use it (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in).